src/main/scala/com/microsoft/hyperspace/index/sources/interfaces.scala

```diff
@@ -89,6 +89,14 @@ trait FileBasedSourceProvider extends SourceProvider {
    */
   def refreshRelation(relation: Relation): Option[Relation]
 
+  /**
+   * Returns a file format name to read partial data for a given [[Relation]].
+   *
+   * @param relation [[Relation]] object to read partial data files.
+   * @return File format to read partial data files.
+   */
+  def partialReadFileFormat(relation: Relation): Option[String]
```

Comment:
I think this should be `def internalFileFormatName` or something to denote the actual file format used for storing the files?